### ISSUE TITLE
`no-null`: allow in ternary if other part contains null

### ISSUE THREAD

**Issue by kkmuffme** [sindresorhus/eslint-plugin-unicorn#2438]
### Description

e.g.

```
const myElement = foo === true ? document.getElementById( 'abc' ) : null;
```

Using `undefined` here instead of null makes the code worse in my opinion.
If there's a ternary where one hand can contain null, the other part shouldn't report an error when null is used

Since you don't have type info in eslint, that would obviously only be for native JS functions (and/or make it configurable).

### Fail

same as now

### Pass

```js
const myElement = window.location.hash ? document.getElementById( 'abc' ) : null;
```

**Comment by sindresorhus** [comment]
The intent of the rule is to not use `null`. I don't want to start adding exceptions as there are too many potential ones.
